src/ngraph/op/dequantize.cpp

```diff
@@ -50,9 +50,9 @@ void op::Dequantize::validate_and_infer_types()
     NODE_VALIDATION_ASSERT(this, m_type.is_real()) << "Output element type (" << m_type
                                                    << ") must be a floating point number";
 
-    NODE_VALIDATION_ASSERT(this, get_input_element_type(SCALE).is_real())
+    NODE_VALIDATION_ASSERT(this, get_input_element_type(SCALE) == m_type)
```

Comment:
This is the change I alluded to in the PR description.

Reply:
My original PR has == m_type (instead of is_real) and @diyessi asked me to change.  I like it better this way, TBH.